### PR TITLE
soa optimizaion for release branch

### PR DESCRIPTION
**Pull request by dmitry741**
SOA optimization for releases branch.
AVX512 architecture.

### CODE REVIEW

**Comment by dmitry741** [comment]
make test

File: include/data_management/data/soa_numeric_table.h [hunk 1]
```diff
@@ -380,10 +401,82 @@ class DAAL_EXPORT SOANumericTable : public NumericTable
...
+
+        return (int)f0.indexType == (int)internal::getConversionDataType<float>()
+               || (int)f0.indexType == (int)internal::getConversionDataType<double>();
+    }
+
     template <typename T>
     services::Status getTBlock(size_t idx, size_t nrows, ReadWriteMode rwFlag, BlockDescriptor<T> & block)
     {
-        size_t ncols = getNumberOfColumns();
+size_t ncols = getNumberOfColumns();
```
Comment (ShvetsKS) [inline include/data_management/data/soa_numeric_table.h]: add alignment

File: service/kernel/data_management/data_conversion_cpu.cpp [hunk 2]
```diff
@@ -25,6 +25,99 @@ namespace data_management
...
+    char const * ptrByte = (char *)ptrMin;
+
+    const size_t nColSize = ncols - ncols % 8;
+
+    for (size_t i = 0; i < nrows; ++i)
+    {
+        for (size_t j = 0; j < nColSize; j += 8)
+        {
+            __m512d p = _mm512_i64gather_pd(*((__m512i *)&arrOffsets[j]), pmin + i, 1);
+            _mm512_storeu_pd(pd + i * ncols + j, p);
```
Comment (ShvetsKS) [inline service/kernel/data_management/data_conversion_cpu.cpp]: This instruction was not defined with usage icc180. Problem happened on CEV build of 2020 mnt with icc180 compiler. Please check it.

Comment (dmitry741) [reply service/kernel/data_management/data_conversion_cpu.cpp]: It's release branch

Comment (ShvetsKS) [reply service/kernel/data_management/data_conversion_cpu.cpp]: the same code was merged to rls/daal-2020-mnt if I`m not mistaken.

Comment (dmitry741) [reply service/kernel/data_management/data_conversion_cpu.cpp]: yes

Comment (dmitry741) [reply service/kernel/data_management/data_conversion_cpu.cpp]: according to logs build for icc180 is successful

File: include/data_management/data/soa_numeric_table.h (hunk 1 above)
Comment (dmitry741) [reply include/data_management/data/soa_numeric_table.h]: done

**Pull request merged**